rust/src/io/object_store.rs

```diff
@@ -176,6 +176,20 @@ impl ObjectStore {
         ObjectWriter::new(self, path).await
     }
 
+    /// Read a directory (start from base directory) and returns all sub-paths in the directory.
+    pub async fn read_dir(&self, dir_path: impl Into<Path>) -> Result<Vec<String>> {
```

Comment:
Should this return `Result<Vec<Path>>` instead for downstream processing?

Reply:
Ok, will fix it.